electrum/lnsweep.py

```diff
@@ -297,11 +297,13 @@ def create_txns_for_htlc(
         ctn=ctn)
     for (direction, htlc), (ctx_output_idx, htlc_relative_idx) in htlc_to_ctx_output_idx_map.items():
         if direction == RECEIVED:
-            if chan.lnworker.get_payment_status(htlc.payment_hash) == PR_PAID:
-                preimage = chan.lnworker.get_preimage(htlc.payment_hash)
-            else:
+            if chan.lnworker.is_incomplete_mpp(htlc.payment_hash):
```

Comment:
If the resolution is `EXPIRED` or `FAILED`, and we have the preimage, this will release it.
I am not sure if that is safe to do.
Is that an intended behaviour change on this line?

I mean, just consider lnworker.stopping_soon. That can transition the mpp to EXPIRED, and then someone force-closes, and we have the preimage (e.g. because this is a simple invoice we created), then we release it onchain even for a partial payment?

Reply:
oh indeed. that was not intended.